solution.cpp

```diff
@@ -1,23 +1,74 @@
-/* ----------------------------------------------------------------------------
- * Copyright &copy; 2015 TODO(your name) <TODO(your email)>
- * Released under the [MIT License] (http://opensource.org/licenses/MIT)
- * ------------------------------------------------------------------------- */
-
-/**
- * TODO: program description
- */
-
-// TODO: `#include`s and function prototypes
-
-// ----------------------------------------------------------------------------
-
-int main() {
-    // TODO
-
-    return 0;  // success
-}
-
-// ----------------------------------------------------------------------------
-
-// TODO: other functions
-
++/*-----------------------------------------------------------------------
++ *              Name: Thanh Phan
++ *              Course: CPSC 121, Tue Lab
++ *              Professor: Ben Blazak
++ * -----------------------------------------------------------------------*/
++/* -----------------------------------------------------------------------
++ * Copyright &copy; 2015 Thanh Phan <tphan2650@gmail.com
++ * Released under the [MIT License] (http://opensource.org/
++ * licenses/MIT)
++ * -----------------------------------------------------------------------*/
++
++
++#include <iostream>
++#include <string>
++using std::string;
++using std::cin;
++using std::cout;
++using std::endl;
++
++//Function to convert integer to Roman numerals
++string int_to_roman(int n)
++{
++    string roman_num;
++    int o, t, h, th;
++    
++    //Array of Roman Numerals to find which integer 
++    //was entered and replace it with the corresponding array element
++    const char*const ones [] = {"", "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX"};
++    
++   const char* const tens [] = {"", "X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC"};
++    
++    const char* const hundreds[] = {"", "C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM"};
++    
++    const char* const thousands [] = {"", "M", "MM", "MMM", "MMMM"};
++    
++    if (n < 5000) {
++    th = n / 1000;
++    n = n % 1000; //Removes the thousands place and uses the remainder
++    
++    h=n/100;
++    n=n%100; //Removes the hundreds place and use the remainder
++    
++    t=n/10;  
++   o=n%10;
++    roman_num = roman_num+ thousands[th]+hundreds[h]+ tens[t]+ones[o]; //Adds the roman numerals together
++    }
++    
++    
++    else if (n>=5000) //Tells user if the number is too high to convert
++    {
++        cout << endl;
++        cout << "I'm not smart enough to do that, go lower!" << endl;
++    }
++    cout << roman_num << endl << endl;
```

Comment:
it looks like line 54 should be below line 45, because otherwise `roman_num` might be uninitialized when this line executes